packages/devtools_app/lib/src/inspector/flutter/inspector_data_models.dart

```diff
@@ -102,16 +103,23 @@ class LayoutProperties {
             : node?.childrenNow
                 ?.map((child) =>
                     LayoutProperties(child, copyLevel: copyLevel - 1))
-                ?.toList(growable: false);
+                ?.toList(growable: false) {
+    for (var child in children) {
+      child.parent = this;
+    }
+  }
 
+  LayoutProperties parent;
   final RemoteDiagnosticsNode node;
   final List<LayoutProperties> children;
   final BoxConstraints constraints;
   final String description;
-  final int flexFactor;
+  int flexFactor;
```

Comment:
Changes to LayoutProperties will be easier to animate if all the fields are final.

We can create a Tween<LayoutProperties> that linearly interpolates (lerps) the constraints, flexfactor, and size, and drive the tween with an animation.

Is it feasible to create new layout properties instead of editing them?

Reply:
Yes it is because in the end after changing the flex factor I will re-fetch the node anyway, but the change in the rendering would not be instant (have to wait for the fetch to finish first and call setState later)